madprops/serializers.py

```diff
@@ -61,95 +91,160 @@ class Property(model):
 
     _options_class = PropertySerializerOptions
 
-    def field_to_native(self, obj, field_name):
-        if obj is None:
-            return None
+    def __init__(self, *args, **kwargs):
+        super(PropertySerializer, self).__init__(*args, **kwargs)
+        self.opts = self._options_class(self.Meta)
+        self.Meta.list_serializer_class = ListToDictSerializer
 
-        return self._to_representation(getattr(obj, field_name).all())
-
-    @cached_property
-    def data(self):
-        return self._to_representation(self.objects)
-
-    def _to_representation(self, objects):
-        return dict((obj.name, self._get_value(obj)) for obj in objects)
+    def to_representation(self, obj):
+        return {obj.name: self._get_value(obj)}
 
     def _get_value(self, obj):
```

Comment:
Since this method is used only in `to_representation` we can move all stuff there.